uprotocol/uri/factory/uresource_builder.py

```diff
@@ -25,18 +25,28 @@
 # -------------------------------------------------------------------------
 
 
+import re
 from uprotocol.proto.uri_pb2 import UResource
+from multimethod import multimethod
+from typing import Union
 
 
 class UResourceBuilder:
     MAX_RPC_ID = 1000
 
+    # The minimum topic ID, below this value are methods.
+    MIN_TOPIC_ID = 0x8000
+
     @staticmethod
     def for_rpc_response():
         return UResource(name="rpc", instance="response", id=0)
 
-    @staticmethod
-    def for_rpc_request(method, id=None):
+    @multimethod
+    def for_rpc_request(method: Union[str, None]):
+        return UResourceBuilder.for_rpc_request(method, None)
+
+    @multimethod
+    def for_rpc_request(method: Union[str, None], id: Union[int, None] = None):
```

Comment:
Specifying None in type hints is often redundant as Python allows None by default for any parameter.